---
title: "Increment_genome_plots"
author: "Lewis Fisher"
date: "12/04/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
 




```{r libraries}
# remove.packages("ComplexHeatmap")
# 
# detach("package:circlize", unload = TRUE)
# remove.packages("circlize")
# install_version("circlize", "0.4.10")
# install.packages("devtools")
# library(devtools)
# install_github("jokergoo/ComplexHeatmap")
# if (!requireNamespace("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# BiocManager::install("ComplexHeatmap")

# install.packages("ggplot2")

# install_github("jokergoo/circlize")
library(reshape)
library(ggplot2)
library(circlize)
library(ComplexHeatmap)
library(stringr)
library(dplyr)
library(gtools)
packageVersion("circlize")

```

```{r import data}
cov_dir<- "C:\\Users\\Lewis\\OneDrive - The University of Liverpool\\PhD project - pseudomonas resistance\\Genomics\\increment_increase_evolution\\coverage_plots"
# cov_dir<- "C:\\Users\\lew619\\OneDrive - The University of Liverpool\\PhD project - pseudomonas resistance\\Genomics\\increment_increase_evolution\\coverage_plots"
# cov_dir<- "C:\\Users\\lew61\\OneDrive - The University of Liverpool\\PhD project - pseudomonas resistance\\Genomics\\increment_increase_evolution\\coverage_plots"
setwd(cov_dir)
list.files()

cov_data<- read.csv("increment_coverage_medians.bed", header = F, sep = "\t")


########################
# mut_dir<- "C:\\Users\\lew619\\OneDrive - The University of Liverpool\\PhD project - pseudomonas resistance\\Genomics\\increment_increase_evolution\\increment_gd_sub_files"
mut_dir<- "C:\\Users\\Lewis\\OneDrive - The University of Liverpool\\PhD project - pseudomonas resistance\\Genomics\\increment_increase_evolution\\increment_gd_sub_files"
# mut_dir<- "C:\\Users\\lew61\\OneDrive - The University of Liverpool\\PhD project - pseudomonas resistance\\Genomics\\increment_increase_evolution\\increment_gd_sub_files"
setwd(mut_dir)
list.files()

mut_data<- read.csv("Increment_breseq_sub_reannotation.csv", header = T, sep = ",")
mut_data$title = str_remove(mut_data$title, "_breseq")
mut_data$replicate<- str_split_fixed(mut_data$title, "_", 2)[,1]
mut_data$transfer<- str_split_fixed(mut_data$title, "_", 2)[,2]
mut_data$condition = str_extract(mut_data$replicate, "(EDTA)|(NAB)|(NE)|(TSB)")

mut_data<- subset(mut_data, snp_type != "intergenic")
unique(mut_data$type)


combined_annotation<- c()
for (i in 1:nrow(mut_data)) {
  enog_gene<- mut_data$eggnog_gene[i]
  gene_name<- mut_data$gene_name[i]
  if (enog_gene == "unknown" | enog_gene == "-") {
     combined_annotation<- append(combined_annotation,gene_name)
    }else {combined_annotation<- append(combined_annotation, enog_gene)
  }
}
length(combined_annotation)
nrow(mut_data)
mut_data$combined_annotation = combined_annotation

```



```{r heatmap}

# https://github.com/spencer411/2021_Bioinformatics_Fellowship/blob/main/presence_absence_heatmap.R

# populate a binary table of presence/absence data for unique lists of title and gene names

mut_data.highfreq<- subset(mut_data, frequency > 0.05)


title_len<- length(unique(mut_data.highfreq$title))
title_name<- unique(mut_data.highfreq$title)

# gene_length<- length(unique(mut_data.highfreq$gene_name))
# gene_name<- unique(mut_data.highfreq$gene_name)

gene_length<- length(unique(mut_data.highfreq$combined_annotation))
gene_name<- unique(mut_data.highfreq$combined_annotation)

# blank_matrix<- matrix(rep(0,title_len * gene_length),
#            nrow = title_len, ncol = gene_length, dimnames = list(title_name, gene_name))
# count_matrix<- blank_matrix
blank_matrix<- matrix(rep(0,title_len * gene_length),
           nrow = title_len, ncol = gene_length, dimnames = list(title_name, gene_name))
count_matrix<- blank_matrix

for_i<- rownames(blank_matrix)
for_e<- colnames(blank_matrix)
for (i in 1:length(for_i)) {
  
  cur_title<- for_i[i]
  cur_sub<- subset(mut_data.highfreq, title == cur_title)
  for (e in 1:length(for_e)) {
    
    cur_gene<- for_e[e]
    gene_count<- sum(cur_sub$combined_annotation == cur_gene)
    if (gene_count >= 1) {
      blank_matrix[i,e]<- 1
      count_matrix[i,e]<- gene_count
    }
  }
}

frame_data<- data.frame(count_matrix)
melt.data<-melt(blank_matrix)
colnames(melt.data)<- c("title", "Gene", "value")
melt.data.count<-melt(count_matrix)
colnames(melt.data.count)<- c("title", "Gene", "value")

# pattern <- "(T14)$"
# 
# words_extracted <- str_subset(melt.data$title, pattern)
# 
# melt.data<- melt.data[melt.data$title %in% unique(words_extracted),]
Gene_list<- unique(as.matrix(melt.data$Gene))
for (i in 1:length(unique(melt.data$Gene))) {
  gene_i<- Gene_list[i]
  df_cur<- subset(melt.data, Gene == gene_i)
  if (sum(df_cur$value) == 0){
    melt.data<- subset(melt.data, Gene != gene_i)
  } 

}
# qplot(data=melt.data,
#       x=Gene,
#       y=title,
#       fill=factor(value),
#       geom="tile")+scale_fill_manual(values=c("0"="white", "1"="red")) +
#       theme(text = element_text(size=10),
#       axis.text.y = element_text(size=4),
#       axis.text.x = element_text(size=4, angle=90, hjust=1))


ggplot(melt.data.count,aes( title, Gene,fill=value))+
  geom_tile(aes(fill = value), colour = "white") +
  scale_fill_gradient(low = "white", high = "red") +
  theme(axis.text.x = element_text(size=6, angle = 90),
        axis.text.y = element_text(size = 6))
  
ggsave("Heatmap_test.png", plot = last_plot(), path = mut_dir, scale = 2, width = 4, height = 8, units = c("in"), dpi = 600)

```

```{r mutation data}


# EDTA1<- subset(cov_data, V1 == "EDTA1 T1" | V1 == "EDTA1 T7" | V1 == "EDTA1 T14")
EDTA<- subset(cov_data, V1 == "EDTA1 T1" | V1 == "EDTA2 T1" | V1 == "EDTA3 T1" | V1 == "EDTA4 T1")
sample_factors<- str_extract(unique(EDTA$V1), "(EDTA[1-4]{1})")
sectors<- c("EDTA1", "EDTA2", "EDTA3", "EDTA4")
sample_factors<- factor(sample_factors, levels = sectors)

# Obtaining the genome size of each sample
sample_max_EDTA<- c()
for (i in 1:length(unique(EDTA$V1))) {
  sample_list<- unique(EDTA$V1)
  sample_sub<- subset(EDTA, V1 == sample_list[i])
  listtmp<- max(sample_sub$V3)
  
  sample_max_EDTA<- append(sample_max_EDTA, listtmp)
}
 
# mut_all<-  mut_all[c("replicate", "position_start","position_end","frequency", "gene_name", "type")]
# colnames(mut_all)= c("chr", "start", "end", "value1", "gene", "type")

# png("EDTA1_coverage_plot.png", res = 300, height = 4, width = 4, units = "in") 

mut_EDTA<- subset(mut_data, condition == "EDTA")
# mut_EDTA<-  mut_EDTA[c("replicate", "position_start","position_end","frequency", "gene_name", "type", "transfer")]
mut_EDTA<-  mut_EDTA[c("replicate", "position_start","position_end","frequency", "combined_annotation", "type", "transfer")]

colnames(mut_EDTA)= c("chr", "start", "end", "value1", "gene", "type", "transfer")
unique(mut_EDTA$type)
mut_dir
png(paste(mut_dir,"\\EDTA_mut_plot.png", sep = ""), res = 1200, height = 8, width = 8, units = "in") 

circos.clear()

ref<- matrix(c(rep(0, length(sample_factors) ), sample_max_EDTA), ncol=2)
circos.par("track.height"=0.8, "gap.degree"= 5.2, cell.padding=c(0, 0, 0, 0), "start.degree" = -182.6)

circos.initialize(factors = sample_factors, xlim=ref)
col_text <- "grey40"

# circos.genomicLabels(mut_all[,c("chr", "start", "gene")], labels = mut_all$gene, cex=0.2, side="outside", labels_height=0.1, padding = 1, niceFacing = TRUE, connection_height = 0.1, line_lwd = 0.1)
circos.genomicLabels(mut_EDTA, labels= mut_EDTA$gene, cex=0.2, side="outside", labels_height=0.1, padding = 0.5, niceFacing = TRUE, connection_height = 0.1, line_lwd = 0.1, col = mut_EDTA$type %>% str_replace_all(c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1")), line_col = mut_EDTA$type %>% str_replace_all(c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1")))

# Options
track_size_mut<- 0.03
border_colours<- c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1")


circos.genomicTrack(data=subset(mut_EDTA, transfer == "T14"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T14", sector.index= "EDTA1", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

circos.genomicTrack(data= subset(mut_EDTA, transfer == "T7"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T7", sector.index= "EDTA1",cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

circos.genomicTrack(data=subset(mut_EDTA, transfer == "T1"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T1", sector.index= "EDTA1", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

# genomes
circos.track(ylim=c(0, 1), panel.fun=function(x, y) {
chr=CELL_META$sector.index
xlim=CELL_META$xlim
ylim=CELL_META$ylim
circos.text(mean(xlim), mean(ylim), chr, cex=1, col=col_text, 
facing="bending.inside", niceFacing=TRUE)
}, bg.col="slategray3", bg.border=F, track.height=0.06)

brk <- c(0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5, 5.5, 6, 6.3)*10^6
circos.track(track.index = get.current.track.index(), panel.fun=function(x, y) {
circos.axis(h="bottom", major.at=brk, labels=round(brk/10^6, 1), labels.cex=0.4, 
col=col_text, labels.col=col_text, lwd=0.7, labels.facing="outside", direction = "inside")
circos.text(mean(CELL_META$xlim), -0.8, "Size (MB)", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)
}, bg.border=F)



lgd = Legend(at = c("SNP","INS","DEL"), title = "Mutation Type", legend_gp = gpar(fill = c("darkorchid2","chartreuse3","darkorange1")))

lgd_list_vertical = packLegend(lgd)
draw(lgd_list_vertical, x = unit(4, "mm"), y = unit(4, "mm"), just = c("left", "bottom"))
# labelling the genome
# snp_nonsyn
#line_col="grey80",

dev.off()


```



```{r mutation data}

# EDTA1<- subset(cov_data, V1 == "EDTA1 T1" | V1 == "EDTA1 T7" | V1 == "EDTA1 T14")
NAB<- subset(cov_data, V1 == "NAB1 T1" | V1 == "NAB2 T1" | V1 == "NAB3 T1" | V1 == "NAB4 T1")
sample_factors<- str_extract(unique(NAB$V1), "(NAB[1-4]{1})")
sectors<- c("NAB1", "NAB2", "NAB3", "NAB4")
sample_factors<- factor(sample_factors, levels = sectors)




 # sort(sample_factors, decreasing = TRUE)
# Obtaining the genome size of each sample
sample_max_NAB<- c()
for (i in 1:length(unique(NAB$V1))) {
  sample_list<- unique(NAB$V1)
  sample_sub<- subset(NAB, V1 == sample_list[i])
  listtmp<- max(sample_sub$V3)
  
  sample_max_NAB<- append(sample_max_NAB, listtmp)
}

# mut_all<-  mut_all[c("replicate", "position_start","position_end","frequency", "gene_name", "type")]
# colnames(mut_all)= c("chr", "start", "end", "value1", "gene", "type")



mut_NAB<- subset(mut_data, condition == "NAB")
c("replicate", "position_start","position_end","frequency", "combined_annotation", "type", "transfer")
# mut_NAB<-  mut_NAB[c("replicate", "position_start","position_end","frequency", "gene_name", "type", "transfer")]
mut_NAB<-  mut_NAB[c("replicate", "position_start","position_end","frequency", "combined_annotation", "type", "transfer")]

colnames(mut_NAB)= c("chr", "start", "end", "value1", "gene", "type", "transfer")
unique(mut_NAB$type)
png("C:\\Users\\Lewis\\OneDrive - The University of Liverpool\\PhD project - pseudomonas resistance\\Genomics\\increment_increase_evolution\\increment_gd_sub_files\\NAB_mut_plot.png", res = 1200, height = 8, width = 8, units = "in") 

circos.clear()

ref<- matrix(c(rep(0, length(sample_factors) ), sample_max_NAB), ncol=2)
circos.par("track.height"=0.8, "gap.degree"= 5.2, cell.padding=c(0, 0, 0, 0), "start.degree" = -182.6)

circos.initialize(factors = sample_factors, xlim=ref)
col_text <- "grey40"

# circos.genomicLabels(mut_all[,c("chr", "start", "gene")], labels = mut_all$gene, cex=0.2, side="outside", labels_height=0.1, padding = 1, niceFacing = TRUE, connection_height = 0.1, line_lwd = 0.1)
circos.genomicLabels(mut_NAB, labels= mut_NAB$gene, cex=0.2, side="outside", labels_height=0.1, padding = 0.5, niceFacing = TRUE, connection_height = 0.1, line_lwd = 0.1, col = mut_NAB$type %>% str_replace_all(c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1")), line_col = mut_NAB$type %>% str_replace_all(c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1")))
#
track_size_mut<- 0.03
border_colours<- c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1")

# mut_test<- value$type %>% str_replace_all(border_colours)
# 

circos.genomicTrack(data=subset(mut_NAB, transfer == "T14"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T14", sector.index= "NAB1", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

circos.genomicTrack(data= subset(mut_NAB, transfer == "T7"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T7", sector.index= "NAB1",cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

circos.genomicTrack(data=subset(mut_NAB, transfer == "T1"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T1", sector.index= "NAB1", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

# genomes
circos.track(ylim=c(0, 1), panel.fun=function(x, y) {
chr=CELL_META$sector.index
xlim=CELL_META$xlim
ylim=CELL_META$ylim
circos.text(mean(xlim), mean(ylim), chr, cex=1, col=col_text, 
facing="bending.inside", niceFacing=TRUE)
}, bg.col="slategray3", bg.border=F, track.height=0.06)

brk <- c(0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5, 5.5, 6, 6.3)*10^6
circos.track(track.index = get.current.track.index(), panel.fun=function(x, y) {
circos.axis(h="bottom", major.at=brk, labels=round(brk/10^6, 1), labels.cex=0.4, 
col=col_text, labels.col=col_text, lwd=0.7, labels.facing="outside", direction = "inside")
circos.text(mean(CELL_META$xlim), -0.8, "Size (MB)", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)
}, bg.border=F)



lgd = Legend(at = c("SNP","INS","DEL"), title = "Mutation Type", legend_gp = gpar(fill = c("darkorchid2","chartreuse3","darkorange1")))

lgd_list_vertical = packLegend(lgd)
draw(lgd_list_vertical, x = unit(4, "mm"), y = unit(4, "mm"), just = c("left", "bottom"))
# labelling the genome
# snp_nonsyn
#line_col="grey80",

dev.off()


```

```{r mutation data}

# EDTA1<- subset(cov_data, V1 == "EDTA1 T1" | V1 == "EDTA1 T7" | V1 == "EDTA1 T14")
NE<- subset(cov_data, V1 == "NE1 T1" | V1 == "NE2 T1" | V1 == "NE3 T1" | V1 == "NE4 T1")
sample_factors<- str_extract(unique(NE$V1), "(NE[1-4]{1})")
sectors<- c("NE1", "NE2", "NE3", "NE4")
sample_factors<- factor(sample_factors, levels = sectors)




 # sort(sample_factors, decreasing = TRUE)
# Obtaining the genome size of each sample
sample_max_NE<- c()
for (i in 1:length(unique(NE$V1))) {
  sample_list<- unique(NE$V1)
  sample_sub<- subset(NE, V1 == sample_list[i])
  listtmp<- max(sample_sub$V3)
  
  sample_max_NE<- append(sample_max_NE, listtmp)
}

# mut_all<-  mut_all[c("replicate", "position_start","position_end","frequency", "gene_name", "type")]
# colnames(mut_all)= c("chr", "start", "end", "value1", "gene", "type")



mut_NE<- subset(mut_data, condition == "NE")
# mut_NE<-  mut_NE[c("replicate", "position_start","position_end","frequency", "gene_name", "type", "transfer")]
mut_NE<-  mut_NE[c("replicate", "position_start","position_end","frequency", "combined_annotation", "type", "transfer")]

colnames(mut_NE)= c("chr", "start", "end", "value1", "gene", "type", "transfer")
unique(mut_NE$type)
png("C:\\Users\\Lewis\\OneDrive - The University of Liverpool\\PhD project - pseudomonas resistance\\Genomics\\increment_increase_evolution\\increment_gd_sub_files\\NE_mut_plot.png", res = 1200, height = 8, width = 8, units = "in") 

circos.clear()

ref<- matrix(c(rep(0, length(sample_factors) ), sample_max_NE), ncol=2)
circos.par("track.height"=0.8, "gap.degree"= 5.2, cell.padding=c(0, 0, 0, 0), "start.degree" = -182.6)

circos.initialize(factors = sample_factors, xlim=ref)
col_text <- "grey40"

# circos.genomicLabels(mut_all[,c("chr", "start", "gene")], labels = mut_all$gene, cex=0.2, side="outside", labels_height=0.1, padding = 1, niceFacing = TRUE, connection_height = 0.1, line_lwd = 0.1)
circos.genomicLabels(mut_NE, labels= mut_NE$gene, cex=0.2, side="outside", labels_height=0.1, padding = 0.5, niceFacing = TRUE, connection_height = 0.1, line_lwd = 0.1, col = mut_NE$type %>% str_replace_all(c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1")), line_col = mut_NE$type %>% str_replace_all(c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1")))
#
track_size_mut<- 0.03
border_colours<- c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1")

# mut_test<- value$type %>% str_replace_all(border_colours)
# 

circos.genomicTrack(data=subset(mut_NE, transfer == "T14"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T14", sector.index= "NE1", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

circos.genomicTrack(data= subset(mut_NE, transfer == "T7"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T7", sector.index= "NE1",cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

circos.genomicTrack(data=subset(mut_NE, transfer == "T1"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T1", sector.index= "NE1", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

# genomes
circos.track(ylim=c(0, 1), panel.fun=function(x, y) {
chr=CELL_META$sector.index
xlim=CELL_META$xlim
ylim=CELL_META$ylim
circos.text(mean(xlim), mean(ylim), chr, cex=1, col=col_text, 
facing="bending.inside", niceFacing=TRUE)
}, bg.col="slategray3", bg.border=F, track.height=0.06)

brk <- c(0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5, 5.5, 6, 6.3)*10^6
circos.track(track.index = get.current.track.index(), panel.fun=function(x, y) {
circos.axis(h="bottom", major.at=brk, labels=round(brk/10^6, 1), labels.cex=0.4, 
col=col_text, labels.col=col_text, lwd=0.7, labels.facing="outside", direction = "inside")
circos.text(mean(CELL_META$xlim), -0.8, "Size (MB)", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)
}, bg.border=F)



lgd = Legend(at = c("SNP","INS","DEL"), title = "Mutation Type", legend_gp = gpar(fill = c("darkorchid2","chartreuse3","darkorange1")))

lgd_list_vertical = packLegend(lgd)
draw(lgd_list_vertical, x = unit(4, "mm"), y = unit(4, "mm"), just = c("left", "bottom"))
# labelling the genome
# snp_nonsyn
#line_col="grey80",

dev.off()


```


```{r mutation data}

# EDTA1<- subset(cov_data, V1 == "EDTA1 T1" | V1 == "EDTA1 T7" | V1 == "EDTA1 T14")
TSB<- subset(cov_data, V1 == "TSB1 T1" | V1 == "TSB2 T1" | V1 == "TSB3 T1" | V1 == "TSB4 T1")
sample_factors<- str_extract(unique(TSB$V1), "(TSB[1-4]{1})")
sectors<- c("TSB1", "TSB2", "TSB3", "TSB4")
sample_factors<- factor(sample_factors, levels = sectors)




 # sort(sample_factors, decreasing = TRUE)
# Obtaining the genome size of each sample
sample_max_TSB<- c()
for (i in 1:length(unique(TSB$V1))) {
  sample_list<- unique(TSB$V1)
  sample_sub<- subset(TSB, V1 == sample_list[i])
  listtmp<- max(sample_sub$V3)
  
  sample_max_TSB<- append(sample_max_TSB, listtmp)
}

# mut_all<-  mut_all[c("replicate", "position_start","position_end","frequency", "gene_name", "type")]
# colnames(mut_all)= c("chr", "start", "end", "value1", "gene", "type")



mut_TSB<- subset(mut_data, condition == "TSB")
# mut_TSB<-  mut_TSB[c("replicate", "position_start","position_end","frequency", "gene_name", "type", "transfer")]
mut_TSB<-  mut_TSB[c("replicate", "position_start","position_end","frequency", "combined_annotation", "type", "transfer")]

colnames(mut_TSB)= c("chr", "start", "end", "value1", "gene", "type", "transfer")
unique(mut_TSB$type)
png("C:\\Users\\Lewis\\OneDrive - The University of Liverpool\\PhD project - pseudomonas resistance\\Genomics\\increment_increase_evolution\\increment_gd_sub_files\\TSB_mut_plot.png", res = 1200, height = 8, width = 8, units = "in") 

circos.clear()

ref<- matrix(c(rep(0, length(sample_factors) ), sample_max_TSB), ncol=2)
circos.par("track.height"=0.8, "gap.degree"= 5.2, cell.padding=c(0, 0, 0, 0), "start.degree" = -182.6)

circos.initialize(factors = sample_factors, xlim=ref)
col_text <- "grey40"

# circos.genomicLabels(mut_all[,c("chr", "start", "gene")], labels = mut_all$gene, cex=0.2, side="outside", labels_height=0.1, padding = 1, niceFacing = TRUE, connection_height = 0.1, line_lwd = 0.1)
circos.genomicLabels(mut_TSB, labels= mut_TSB$gene, cex=0.2, side="outside", labels_height=0.1, padding = 0.5, niceFacing = TRUE, connection_height = 0.1, line_lwd = 0.1, col = mut_TSB$type %>% str_replace_all(c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1", "AMP" = "red")), line_col = mut_TSB$type %>% str_replace_all(c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1", "AMP" = "red")))
#
track_size_mut<- 0.03
border_colours<- c("SNP" = "darkorchid2", "INS" = "chartreuse3", "DEL" = "darkorange1", "AMP" = "red")

# mut_test<- value$type %>% str_replace_all(border_colours)
# 

circos.genomicTrack(data=subset(mut_TSB, transfer == "T14"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T14", sector.index= "TSB1", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

circos.genomicTrack(data= subset(mut_TSB, transfer == "T7"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T7", sector.index= "TSB1",cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

circos.genomicTrack(data=subset(mut_TSB, transfer == "T1"), panel.fun=function(region, value, ...) {
region$start = region$start +8000
circos.genomicRect(region, value, type="l", col= value$type %>% str_replace_all(border_colours), border = NA)
}, track.height=track_size_mut, bg.border=F, bg.col= "gray95")
circos.text(max(CELL_META$xlim) + (0.2*(10^6)), mean(CELL_META$ylim), "T1", sector.index= "TSB1", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)

# genomes
circos.track(ylim=c(0, 1), panel.fun=function(x, y) {
chr=CELL_META$sector.index
xlim=CELL_META$xlim
ylim=CELL_META$ylim
circos.text(mean(xlim), mean(ylim), chr, cex=1, col=col_text, 
facing="bending.inside", niceFacing=TRUE)
}, bg.col="slategray3", bg.border=F, track.height=0.06)

brk <- c(0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5, 5.5, 6, 6.3)*10^6
circos.track(track.index = get.current.track.index(), panel.fun=function(x, y) {
circos.axis(h="bottom", major.at=brk, labels=round(brk/10^6, 1), labels.cex=0.4, 
col=col_text, labels.col=col_text, lwd=0.7, labels.facing="outside", direction = "inside")
circos.text(mean(CELL_META$xlim), -0.8, "Size (MB)", cex=0.6, col=col_text,facing="bending.inside", niceFacing=TRUE)
}, bg.border=F)



lgd = Legend(at = c("SNP","INS","DEL","AMP"), title = "Mutation Type", legend_gp = gpar(fill = c("darkorchid2","chartreuse3","darkorange1", "red")))

lgd_list_vertical = packLegend(lgd)
draw(lgd_list_vertical, x = unit(4, "mm"), y = unit(4, "mm"), just = c("left", "bottom"))
# labelling the genome
# snp_nonsyn
#line_col="grey80",

dev.off()


```
```{r}


```